discovery-infra/download_logs.py

```diff
@@ -62,10 +69,29 @@ def download_logs(client: InventoryClient, cluster: dict, dest: str, must_gather
 
     with suppress(assisted_service_client.rest.ApiException):
         client.download_cluster_events(cluster['id'], os.path.join(output_folder, f"cluster_{cluster['id']}_events.json"))
-        shutil.copy2(os.path.join(os.path.dirname(os.path.realpath(__file__)),"events.html"), output_folder)
+        shutil.copy2(os.path.join(os.path.dirname(os.path.realpath(__file__)), "events.html"), output_folder)
 
     with suppress(assisted_service_client.rest.ApiException):
-        client.download_cluster_logs(cluster['id'], os.path.join(output_folder, f"cluster_{cluster['id']}_logs.tar"))
+        for i in range(MAX_RETRIES):
+            cluster_logs_tar = os.path.join(output_folder, f"cluster_{cluster['id']}_logs.tar")
+
+            with suppress(FileNotFoundError):
+                os.remove(cluster_logs_tar)
+
+            client.download_cluster_logs(cluster['id'], cluster_logs_tar)
+
+            min_number_of_logs = len(cluster['hosts']) + 1 if cluster['status'] == ClusterStatus.INSTALLED else len(cluster['hosts'])
+
+            try:
+                verify_logs_uploaded(cluster_logs_tar, min_number_of_logs, cluster['status'] == ClusterStatus.INSTALLED)
```

Comment:
If you failed on the first try you will fail on all other tries because you don't delete the create dir that holds the extracted files, therefore on each try it will iterate also on old files (that failed in previous try) and fail again.

Reply:
Good catch. Changing implementation to use `tempfile.TemporaryDirectory`